upgrade_policy/quorum_upgrade/sources/quorum_upgrade_policy.move

```diff
@@ -325,6 +379,8 @@ module quorum_upgrade_policy::quorum_upgrade_policy {
             proposer: signer,
         });
 
+        df::remove_if_exists<UpgradeMetadata, VecMap<string::String, vector<u8>>>(&mut proposal.id, UpgradeMetadata {});
+
```

Comment:
Maybe we can also create an `authorize_upgrade_v2` which allows shared object deletion, since we're doing a clean up! 

Could you add it?